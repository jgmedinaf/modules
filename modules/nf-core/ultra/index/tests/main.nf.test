nextflow_process {

    name "Test Process ULTRA_INDEX"
    config "./nextflow.config"
    script "../main.nf"
    process "ULTRA_INDEX"
   
    tag "modules"
    tag "modules_nfcore"
    tag "ultra"
    tag "ultra/index"
    tag "gnu"
    tag "gnu/sort"
   
    test("test_ultra_index") {
        
        setup {
            run("GNU_SORT") {
                script "../../../gnu/sort/main.nf"
                
                process {
                    """
                    input[0] = [
                        [id:'test'],
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.gtf',checkIfExists:true)
                        ]
                    """
                }
            }
        }
        when {
            params {
                gnu_sort_param = '-k1,1 -k4,4n'
                utra_index_param = '--disable_infer'
            }
            process {
                """
                input[0] = file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta',checkIfExists:true)
                input[1] = GNU_SORT.out.sorted.map{it[1]}
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.versions,
                    process.out.index[0][1],
                    process.out.index[0][0].collect{ fileName ->
                        file(fileName.toString()).name }
                ).match() }
            )
        }
    }
}