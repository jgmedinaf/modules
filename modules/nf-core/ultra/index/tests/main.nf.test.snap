{
    "test_ultra_index": {
        "content": [
            [
                "versions.yml:md5,45fdc5637701f0c91e005244cc09a199"
            ],
            "database.db:md5,f9b780bf5a0a5d23256afb99df7137d9",
            [
                "all_splice_pairs_annotations.pickle",
                "all_splice_sites_annotations.pickle",
                "chr_to_id.pickle",
                "exon_choordinates_to_id.pickle",
                "flank_choordinates.pickle",
                "gene_to_small_segments.pickle",
                "id_to_chr.pickle",
                "max_intron_chr.pickle",
                "parts_to_segments.pickle",
                "ref_exon_sequences.pickle",
                "ref_flank_sequences.pickle",
                "ref_part_sequences.pickle",
                "ref_segment_sequences.pickle",
                "refs_id_lengths.pickle",
                "refs_lengths.pickle",
                "segment_id_to_choordinates.pickle",
                "segment_to_gene.pickle",
                "segment_to_ref.pickle",
                "splices_to_transcripts.pickle",
                "transcripts_to_splices.pickle"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.1"
        },
        "timestamp": "2025-05-14T14:50:16.577571936"
    }
}